panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -1093,3 +1080,102 @@ def get_certificates(self) -> dict:
                 result[cert_name] = certificate
 
         return result
+
+    def get_jobs(self) -> dict:
+        """Get details on all jobs.
+
+        This method retrieves all jobs and their details, this means running, pending, finished, etc.
+
+        The actual API command is `show jobs all`.
+
+        # Returns
+
+        dict: All jobs found on the device, indexed by the ID of a job.
+
+        ```python showLineNumbers title="Sample output"
+        {'1': {'description': None,
+            'details': {'line': ['ID population failed',
+                                    'Client logrcvr registered in the middle of a '
+                                    'commit/validate. Aborting current '
+                                    'commit/validate.',
+                                    'Commit failed',
+                                    'Failed to commit policy to device']},
+            'positionInQ': '0',
+            'progress': '100',
+            'queued': 'NO',
+            'result': 'FAIL',
+            'status': 'FIN',
+            'stoppable': 'no',
+            'tdeq': '00:28:32',
+            'tenq': '2023/08/01 00:28:32',
+            'tfin': '2023/08/01 00:28:36',
+            'type': 'AutoCom',
+            'user': None,
+            'warnings': None},
+        '2': {'description': None,
+            'details': {'line': ['Configuration committed successfully',
+                                    'Successfully committed last configuration']},
+            'positionInQ': '0',
+            'progress': '100',
+            'queued': 'NO',
+            'result': 'OK',
+            'status': 'FIN',
+            'stoppable': 'no',
+            'tdeq': '00:28:40',
+            'tenq': '2023/08/01 00:28:40',
+            'tfin': '2023/08/01 00:29:20',
+            'type': 'AutoCom',
+            'user': None,
+            'warnings': None},
+        '3': {'description': None,
+            'details': None,
+            'positionInQ': '0',
+            'progress': '30',
+            'queued': 'NO',
+            'result': 'PEND',
+            'status': 'ACT',
+            'stoppable': 'yes',
+            'tdeq': '00:58:59',
+            'tenq': '2023/08/01 00:58:59',
+            'tfin': None,
+            'type': 'Downld',
+            'user': None,
+            'warnings': None}}
+        ```
+
+        """
+        jobs = self.op_parser(cmd="show jobs all")
+        results = dict()
+
+        for job in jobs["job"]:
+            jid = job["id"]
+            job.pop("id")
+            results[jid] = job
+
+        return results
+
+    def get_update_schedules(self) -> dict:
```

Comment:
Does it also get the job schedules configured on Panorama > Device Deployment **NOT** the Panorama Templates ? 
I'm not sure if this is something pushed to the firewall configuration since that's not on the template so I suspect it might be missing. 

Configured from "Panorama > Device Deployment > Dynamic Updates, click Schedules":
https://docs.paloaltonetworks.com/pan-os/10-1/pan-os-upgrade/upgrade-panorama/deploy-updates-to-firewalls-log-collectors-and-wildfire-appliances-using-panorama/schedule-a-content-update-using-panorama

Reply:
or maybe let's add a helper method that would do this check? it would be separated from the main code, because you need to operate on `pan-os-python` Panorama object. But it could be incorporated into the Ansible module.

Just an idea

Reply:
Yes we can consider this separately. But I think we should put a comment regarding this "Panorama Device Deployment Schedules" are not checked in the method description. Currently, it says that it includes the ones pushed from Panorama but it's not clear about this one.